File: 376-wiggle-subsequence/376-wiggle-subsequence.cpp
class Solution {
public:
    int wiggleMaxLength(vector<int>& nums) 
{
        nums.erase(std::unique(nums.begin(), nums.end()), nums.end());
        if(nums.size() == 1)
            return 1;
        int rlt = 2;
        for(int i=1; i<nums.size()-1; ++i)
        {
            if(nums[i-1] < nums[i] and nums[i+1] < nums[i] or
               nums[i-1] > nums[i] and nums[i+1] > nums[i])
                ++rlt;
        }
        return rlt;
}
};

File: 376-wiggle-subsequence/376-wiggle-subsequence.py
class Solution:
    def wiggleMaxLength(self, nums: List[int]) -> int:
        if(len(nums) < 2): return len(nums)
        sign = 0
        lens = 1
        for i in range(1, len(nums)):
            if(nums[i - 1] < nums[i] and sign != -1):
                lens += 1
                sign = -1
            elif(nums[i - 1] > nums[i] and sign != 1):
                lens += 1
                sign = 1
        return lens
                
                    
                
        